scripts/convert_mochi_to_diffusers.py

```diff
@@ -0,0 +1,307 @@
+        new_state_dict[f"{prefix}block_out.resnets.{i}.norm1.norm_layer.bias"] = original_state_dict[
+            f"blocks.4.{i}.stack.0.bias"
+        ]
+        new_state_dict[f"{prefix}block_out.resnets.{i}.conv1.conv.weight"] = original_state_dict[
+            f"blocks.4.{i}.stack.2.weight"
+        ]
+        new_state_dict[f"{prefix}block_out.resnets.{i}.conv1.conv.bias"] = original_state_dict[
+            f"blocks.4.{i}.stack.2.bias"
+        ]
+        new_state_dict[f"{prefix}block_out.resnets.{i}.norm2.norm_layer.weight"] = original_state_dict[
+            f"blocks.4.{i}.stack.3.weight"
+        ]
+        new_state_dict[f"{prefix}block_out.resnets.{i}.norm2.norm_layer.bias"] = original_state_dict[
+            f"blocks.4.{i}.stack.3.bias"
+        ]
+        new_state_dict[f"{prefix}block_out.resnets.{i}.conv2.conv.weight"] = original_state_dict[
+            f"blocks.4.{i}.stack.5.weight"
+        ]
+        new_state_dict[f"{prefix}block_out.resnets.{i}.conv2.conv.bias"] = original_state_dict[
+            f"blocks.4.{i}.stack.5.bias"
+        ]
+
+    # Convert conv_out (Conv1x1)
+    new_state_dict[f"{prefix}conv_out.weight"] = original_state_dict["output_proj.weight"]
+    new_state_dict[f"{prefix}conv_out.bias"] = original_state_dict["output_proj.bias"]
+
+    return new_state_dict
+
+
+def main(args):
+    if args.dtype is None:
+        dtype = None
+    if args.dtype == "fp16":
+        dtype = torch.float16
+    elif args.dtype == "bf16":
+        dtype = torch.bfloat16
+    elif args.dtype == "fp32":
+        dtype = torch.float32
+    else:
+        raise ValueError(f"Unsupported dtype: {args.dtype}")
+
+    transformer = None
+    vae = None
+
+    if args.transformer_checkpoint_path is not None:
+        converted_transformer_state_dict = convert_mochi_transformer_checkpoint_to_diffusers(
+            args.transformer_checkpoint_path
+        )
+        transformer = MochiTransformer3DModel()
+        transformer.load_state_dict(converted_transformer_state_dict, strict=True)
+        if dtype is not None:
+            transformer = transformer.to(dtype=dtype)
+
+    if args.vae_checkpoint_path is not None:
+        vae = AutoencoderKLMochi(latent_channels=12, out_channels=3)
+        converted_vae_state_dict = convert_mochi_decoder_state_dict_to_diffusers(args.vae_checkpoint_path)
+        vae.load_state_dict(converted_vae_state_dict, strict=True)
+        if dtype is not None:
+            vae = vae.to(dtype=dtype)
+
+    text_encoder_id = "google/t5-v1_1-xxl"
+    tokenizer = T5Tokenizer.from_pretrained(text_encoder_id, model_max_length=TOKENIZER_MAX_LENGTH)
+    text_encoder = T5EncoderModel.from_pretrained(text_encoder_id, cache_dir=args.text_encoder_cache_dir)
+
+    # Apparently, the conversion does not work anymore without this :shrug:
+    for param in text_encoder.parameters():
+        param.data = param.data.contiguous()
+
+    pipe = MochiPipeline(
+        scheduler=FlowMatchEulerDiscreteScheduler(),
```

Comment:
```suggestion
        scheduler=FlowMatchEulerDiscreteScheduler(invert_sigmas=True),
```

Reply:
Hi, pretty thanks for reply, it works! 


https://github.com/user-attachments/assets/13101cfe-a6ca-4c9c-a381-381eeb2ad407